### PR TITLE
[515824] Catch and log exceptions on hover computation

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=515824

**Comment by kthoms** [comment]
Fixes also [514606](https://bugs.eclipse.org/bugs/show_bug.cgi?id=514606)

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hover/html/DefaultEObjectHoverProvider.java [hunk 1]
```diff
@@ -535,7 +535,12 @@ public IInformationControlCreator getHoverControlCreator() {
 
 			@Override
 			public Object getInfo() {
-				return getHoverInfo(object, region, null);
+				try {
+					return getHoverInfo(object, region, null);
+				} catch (RuntimeException e) {
+					LOG.warn("Error on computation of hover information. No hover information available for "+ object + " at " + region, e);
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hover/html/DefaultEObjectHoverProvider.java]: why warn?

Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hover/html/DefaultEObjectHoverProvider.java]: It is typically a Problem in 3rd Party code. The effect is only that no hover doc is shown, besides that it does not influence the system's operability.

Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hover/html/DefaultEObjectHoverProvider.java]: yes but there 3rd party developer might not catch this if it is just logged as warning

**Comment by kthoms** [comment]
Logged now with error level. Please review.

**Comment by cdietrich** [comment]
@svenefftinge @spoenemann can one of your guys have a look too please

**Pull request merged**